homeassistant/components/nmap_tracker/config_flow.py

```diff
@@ -0,0 +1,223 @@
+    """Search adapters for the network."""
+    adapters = ifaddr.get_adapters()
+    local_ip = get_local_ip()
+    network_prefix = (
+        get_ip_prefix_from_adapters(local_ip, adapters) or DEFAULT_NETWORK_PREFIX
+    )
+    return str(ip_network(f"{local_ip}/{network_prefix}", False))
+
+
+def get_ip_prefix_from_adapters(local_ip, adapters):
+    """Find the network prefix for an adapter."""
+    for adapter in adapters:
+        for ip_cfg in adapter.ips:
+            if local_ip == ip_cfg.ip:
+                return ip_cfg.network_prefix
+
+
+def _normalize_ips_and_network(hosts_str):
+    """Check if a list of hosts are all ips or ip networks."""
+
+    normalized_hosts = []
+    hosts = [host for host in cv.ensure_list_csv(hosts_str) if host != ""]
+
+    for host in sorted(hosts):
+        try:
+            start, end = host.split("-", 1)
+            if "." not in end:
+                ip_1, ip_2, ip_3, _ = start.split(".", 3)
+                end = ".".join([ip_1, ip_2, ip_3, end])
+            summarize_address_range(ip_address(start), ip_address(end))
+        except ValueError:
+            pass
+        else:
+            normalized_hosts.append(host)
+            continue
+
+        try:
+            ip_addr = ip_address(host)
+        except ValueError:
+            pass
+        else:
+            normalized_hosts.append(str(ip_addr))
+            continue
+
+        try:
+            network = ip_network(host)
+        except ValueError:
+            return None
+        else:
+            normalized_hosts.append(str(network))
+
+    return normalized_hosts
+
+
+def normalize_input(user_input):
+    """Validate hosts and exclude are valid."""
+    errors = {}
+    normalized_hosts = _normalize_ips_and_network(user_input[CONF_HOSTS])
+    if not normalized_hosts:
+        errors[CONF_HOSTS] = "invalid_hosts"
+    else:
+        user_input[CONF_HOSTS] = ",".join(normalized_hosts)
+
+    normalized_exclude = _normalize_ips_and_network(user_input[CONF_EXCLUDE])
+    if normalized_exclude is None:
+        errors[CONF_EXCLUDE] = "invalid_hosts"
+    else:
+        user_input[CONF_EXCLUDE] = ",".join(normalized_exclude)
+
+    return errors
+
+
+async def _async_build_schema_with_user_input(hass, user_input, include_options):
+    hosts = user_input.get(CONF_HOSTS, await hass.async_add_executor_job(get_network))
+    exclude = user_input.get(
+        CONF_EXCLUDE, await hass.async_add_executor_job(get_local_ip)
+    )
+    schema = {
+        vol.Required(CONF_HOSTS, default=hosts): str,
+        vol.Required(
+            CONF_HOME_INTERVAL, default=user_input.get(CONF_HOME_INTERVAL, 0)
+        ): int,
+        vol.Optional(CONF_EXCLUDE, default=exclude): str,
+        vol.Optional(
+            CONF_OPTIONS, default=user_input.get(CONF_OPTIONS, DEFAULT_OPTIONS)
+        ): str,
+    }
+    if include_options:
+        schema.update(
+            {
+                vol.Optional(
+                    CONF_TRACK_NEW,
```

Comment:
Wait, I don't think we should add this option.

Reply:
Removed, verified reloading still works as well